openeo/metadata.py

```diff
@@ -817,3 +821,101 @@ def get_temporal_dimension(self, stac_obj: pystac.STACObject) -> Union[TemporalD
+        # TODO: "eo:bands" vs "bands" priority based on STAC and EO extension version information
+        summaries = catalog.extra_fields.get("summaries", {})
+        self._warn(f"bands_from_stac_catalog with {summaries.keys()=} (which is non-standard)")
+        if "eo:bands" in summaries:
+            return self._Bands(self._band_from_eo_bands_metadata(b) for b in summaries["eo:bands"])
+        elif "bands" in summaries:
+            return self._Bands(self._band_from_common_bands_metadata(b) for b in summaries["bands"])
+
+        # TODO: instead of warning: exception, or return None?
+        self._warn("bands_from_stac_catalog: no band name source found")
+        return self._Bands([])
+
+    def bands_from_stac_collection(
+        self, collection: pystac.Collection, *, consult_items: bool = True, consult_assets: bool = True
+    ) -> _Bands:
+        # TODO: "eo:bands" vs "bands" priority based on STAC and EO extension version information
+        self._log(f"bands_from_stac_collection with {collection.summaries.lists.keys()=}")
+        if "eo:bands" in collection.summaries.lists:
+            return self._Bands(self._band_from_eo_bands_metadata(b) for b in collection.summaries.lists["eo:bands"])
+        elif "bands" in collection.summaries.lists:
+            return self._Bands(self._band_from_common_bands_metadata(b) for b in collection.summaries.lists["bands"])
+        elif consult_items:
+            bands = self._Bands.merge(
+                self.bands_from_stac_item(item=i, consult_parent=False, consult_assets=consult_assets)
+                for i in collection.get_items()
+            )
+            if bands:
+                return bands
+
+        # TODO: instead of warning: exception, or return None?
+        self._warn("bands_from_stac_collection: no band name source found")
+        return self._Bands([])
+
+    def bands_from_stac_item(
+        self, item: pystac.Item, *, consult_parent: bool = True, consult_assets: bool = True
+    ) -> _Bands:
+        # TODO: "eo:bands" vs "bands" priority based on STAC and EO extension version information
+        self._log(f"bands_from_stac_item with {item.properties.keys()=}")
+        if "eo:bands" in item.properties:
+            return self._Bands(self._band_from_eo_bands_metadata(b) for b in item.properties["eo:bands"])
+        elif "bands" in item.properties:
+            return self._Bands(self._band_from_common_bands_metadata(b) for b in item.properties["bands"])
+        elif consult_parent and (parent_collection := item.get_collection()) is not None:
+            return self.bands_from_stac_collection(collection=parent_collection)
+        elif consult_assets:
+            bands = self._Bands.merge(self.bands_from_stac_asset(asset=a) for a in item.get_assets().values())
+            if bands:
+                return bands
+
+        # TODO: instead of warning: exception, or return None?
+        self._warn("bands_from_stac_item: no band name source found")
```

Comment:
TODO: clean way to not always show these warnings 
e.g. when traversing a collection->item->asset hierarchy and merging band lists, it might be fine that band info is missing at some level

Reply:
moved the `lru_cache` trick to more generic place